micro-rdk/src/esp32/board.rs

```diff
@@ -213,6 +215,11 @@ impl EspBoard {
             };
             (analogs, pins, i2c_confs)
         };
+        //return BoardError if non-analog pin is configured as an analog pin
```

Comment:
Putting the question mark back in L. 187 and putting another one before the semi-colon in L. 190 should make these lines unnecessary. Let me know if it doesn't

Reply:
I see it does let me propagate from collect(), but it leads to some incompatibility type issues I'm trying to fix right now. Thanks! Def better to propagate here then check for an error later on